docs/source/array-design.rst

```diff
@@ -33,6 +36,14 @@ key-value pairs required to eventually compute the desired values, e.g.
     ...
    }
 
+The name of an ``Array`` object can be found in the ``name`` attribute.  One
+can get a nested list of keys with the ``._keys()`` method.  One can flatten
```

Comment:
I've been wondering -- is this actually API? I've always been reluctant to use this method in external code because of the underscore prefix.